### PR TITLE
Followup to EndpointSlices PR for ovn-k node

### PR DESCRIPTION
**Pull request by ricky-rav**
@Pardhakeswar , @girishmg  PTAL

When reviewing https://github.com/ovn-org/ovn-kubernetes/pull/3056, I made a silly mistake and actually left all comments in "pending" state. This is a followup with the changes I had in mind. 

Thanks!

### CODE REVIEW

File: go-controller/pkg/node/healthcheck.go [hunk 1]
```diff
@@ -99,9 +99,8 @@ func countLocalEndpoints(epSlice *discovery.EndpointSlice, nodeName string) int
 	var num int
 	for _, endpoint := range epSlice.Endpoints {
 		if endpoint.NodeName != nil && *endpoint.NodeName == nodeName {
-			continue
+			num++
```
Comment (girishmg) [inline go-controller/pkg/node/healthcheck.go:102]: Good catch!

**Review by girishmg — COMMENTED** [review]
LGTM

File: go-controller/pkg/node/gateway_shared_intf.go [hunk 2]
```diff
@@ -710,22 +710,22 @@ func getEndpointAddresses(endpointSlice *discovery.EndpointSlice) []string {
 func (npw *nodePortWatcher) UpdateEndpointSlice(oldEpSlice, newEpSlice *discovery.EndpointSlice) {
 	oldEpAddr := getEndpointAddresses(oldEpSlice)
 	newEpAddr := getEndpointAddresses(newEpSlice)
+	// should we sort the slices before comparing them? maybe not?
```
Comment (girishmg) [inline go-controller/pkg/node/gateway_shared_intf.go:713]: yes, it's not needed

Comment (ricky-rav) [reply go-controller/pkg/node/gateway_shared_intf.go]: Cool! Let me remove that line... I actually didn't mean to push it, my bad! :)

**Comment by trozet** [comment]
maybe this helps with the failures we are seeing downstream? https://github.com/ovn-org/ovn-kubernetes/issues/3116

**Review by trozet — APPROVED** [review]

**Comment by girishmg** [comment]
please let us know how it goes. thanks. @Pardhakeswar is on vacation.
> maybe this helps with the failures we are seeing downstream? #3116

**Pull request merged**